osbs/build/build_requestv2.py

```diff
@@ -304,6 +304,14 @@ def set_data_from_reactor_config(self, reactor_config_data):
         self.set_source_registry(reactor_config_data)
         self.set_registry_organization(reactor_config_data)
         self.set_required_secrets(reactor_config_data)
+        self.set_build_env(reactor_config_data)
+
+    def set_build_env(self, reactor_config_data):
+        """
+        Propagates build_env variables set in config map to build environment
+        """
+        build_env = reactor_config_data.get('build_env', [])
+        self.template['spec']['strategy']['customStrategy']['env'].extend(build_env)
```

Comment:
:eyes: can we trust that configmap has correct format here?

Reply:
Well it does get validated by schema, I guess I could write this in a way that is more likely to fail if we change the schema

Reply:
It's too late for validation by atomic-rector :)

Reply:
Yeah I realized, should we do some more thorough validation here? Or is it fine the way it is now?

Reply:
I'd say it's fine now